lib/node/index.js

```diff
@@ -843,6 +852,10 @@ Request.prototype._end = function() {
       unzip(req, res);
     }
 
+    if (buffer === undefined && exports.buffer[mime] !== undefined){
```

Comment:
Are you sure it would be a breaking change? This block would only run if a user defined a mime type on `exports.buffer`, and since `exports.buffer` didn't exist until now, previously written code should run as normal. Also, this block still respects the fact that the `buffer()` method takes precedent over everything else.

Reply:
Ah, I see. In that case it's OK. 

However, have you considered https://visionmedia.github.io/superagent/#agents-for-global-state `agent.buffer(true)`?